stable_baselines/common/distributions.py

```diff
@@ -164,9 +165,13 @@ def __init__(self, n_cat):
     def probability_distribution_class(self):
         return CategoricalProbabilityDistribution
 
-    def proba_distribution_from_latent(self, pi_latent_vector, vf_latent_vector, init_scale=1.0, init_bias=0.0):
-        pdparam = linear(pi_latent_vector, 'pi', self.n_cat, init_scale=init_scale, init_bias=init_bias)
-        q_values = linear(vf_latent_vector, 'q', self.n_cat, init_scale=init_scale, init_bias=init_bias)
+    def proba_distribution_from_latent(self, pi_latent_vector, vf_latent_vector, action_mask_vector=None,
+                                       init_scale=1.0, init_bias=0.0):
+        pdparam = apply_action_mask(linear(pi_latent_vector, 'pi', self.n_cat, init_scale=init_scale,
```

Comment:
I'm not sure of what you are doing here.
pdparam corresponds to the logits (before the softmax).
So I would either set them to -inf (if tensorflow allows it, otherwise setting them to a big negative value should do the trick) or set the softmax values to zero afterward.
However, I don't think that setting the logits to zero is a good idea (and this can explain the NaN and that it does not prevent taking invalid actions).

Reply:
So you think setting it to -inf is better?